server/src/main/java/org/elasticsearch/action/admin/indices/rollover/MetadataRolloverService.java

```diff
@@ -150,17 +149,16 @@ private RolloverResult rolloverDataStream(ClusterState currentState, IndexAbstra
         final Version minNodeVersion = currentState.nodes().getMinNodeVersion();
         final DataStream ds = dataStream.getDataStream();
         final IndexMetadata originalWriteIndex = dataStream.getWriteIndex();
-        final String newWriteIndexName = DataStream.getDefaultBackingIndexName(ds.getName(), ds.getGeneration() + 1, minNodeVersion);
-        ds.rollover(new Index(newWriteIndexName, "uuid"), minNodeVersion); // just for validation
-        createIndexService.validateIndexName(newWriteIndexName, currentState); // fails if the index already exists
+        DataStream rolledDataStream = ds.rollover(minNodeVersion); // just for validation
```

Comment:
Maybe remove the comment? Since it is now also be used to get the write index name later in this method?